### PR TITLE
Ports: Add fluidsynth+libmt32emu + activate those dependencies for dosbox-staging

### PR DESCRIPTION
**Pull request by kennethmyhra**
This ports fluidsynth and libmt32emu and activates the support for this in dosbox-staging. Also enables the png dependency so that we can take screenshots in dosbox with CTRL+F5.

The attached videos showcases fluidsynth using a soundfont for Roland SC-55 in the after video.

Sound before:

https://github.com/SerenityOS/serenity/assets/248795/da1d1d4d-e0eb-4b85-94a0-bf4790849c4e

---

Sound after:

https://github.com/SerenityOS/serenity/assets/248795/ec398fb0-e991-454f-ae69-12fe95c905ba

### CODE REVIEW

File: Ports/fluidsynth/package.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    "https://github.com/FluidSynth/fluidsynth/archive/refs/tags/v${version}.tar.gz#f89e8e983ecfb4a5b4f5d8c2b9157ed18d15ed2e36246fa782f18abaea550e0d"
+)
+depends=(
+    'glib'
+)
+
+configure() {
+    cmake \
+      "${configopts[@]}" \
+      -B fluidsynth-build
```
Comment (gmta) [inline Ports/fluidsynth/package.sh:20]: 4 or 2 space indents? :^)

File: Ports/dosbox-staging/package.sh [hunk 2]
```diff
@@ -7,15 +7,15 @@ configopts=(
     '-Ddynamic_core=none'
     '-Dtry_static_libs=opusfile'
     '-Dunit_tests=disabled'
-    '-Duse_fluidsynth=false'
     '-Duse_mt32emu=false'
     '-Duse_opengl=false'
-    '-Duse_png=false'
+    '-Duse_png=true'
```
Comment (gmta) [inline Ports/dosbox-staging/package.sh]: This change is probably irrelevant to this commit, seeing that you even remove it in the next one?

File: Ports/libmt32emu/package.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+   "-DCMAKE_TOOLCHAIN_FILE=${SERENITY_BUILD_DIR}/CMakeToolchain.txt"
+)
+files=(
+    "https://github.com/munt/munt/archive/refs/tags/libmt32emu_${version//./_}.tar.gz#e4524d52d6799a4e32a961a2e92074f14adcb2f110a4e7a06bede77050cfdaf4"
+)
+
+configure() {
+    run cmake \
+      "${configopts[@]}" \
+      -B mt32emu-build
```
Comment (gmta) [inline Ports/libmt32emu/package.sh:19]: Another 2 space indent

**Review by gmta — COMMENTED** [review]
Sweet!

File: Ports/dosbox-staging/package.sh (hunk 2 above)
Comment (kennethmyhra) [reply Ports/dosbox-staging/package.sh]: Ah, yes of course

**Comment by kennethmyhra** [comment]
Addressed review comments

**Review by gmta — APPROVED** [review]

**Pull request merged**